internal/core/core.go

```diff
@@ -495,9 +499,24 @@ func GetSpellingSuggestion[T any](name string, candidates []T, getName func(T) s
 	return bestCandidate
 }
 
-func levenshteinWithMax(s1 []rune, s2 []rune, maxValue float64) float64 {
-	previous := make([]float64, len(s2)+1)
-	current := make([]float64, len(s2)+1)
+func ensureSize(slice []float64, desired int) []float64 {
+	if cap(slice) < desired {
+		return make([]float64, desired)
+	}
+	return slice[:desired]
+}
+
+func levenshteinWithMax(s1 []rune, s2 []rune, maxValue float64, buffers *SpellingSuggestionBuffers) float64 {
+	if buffers == nil {
+		buffers = &SpellingSuggestionBuffers{}
+	}
+
+	buffers.previous = ensureSize(buffers.previous, len(s2)+1)
+	previous := buffers.previous
```

Comment:
This would honestly benefit from simply using a `sync.Pool` to store the buffers globally, rather than scoping the object. Means less plumbing and callers don't have to know anything about it.